Admin/PageAdmin.php

```diff
@@ -351,56 +351,92 @@ protected function configureFormFields(FormMapper $formMapper)
         ;
 
         if ($this->hasSubject() && !$this->getSubject()->isInternal()) {
+            /*
+             * NEXT_MAJOR: remove method_exists and uncomment the second type when dropping sf < 2.8
+             */
             $formMapper
                 ->with('form_page.group_main_label')
-                    ->add('type', 'sonata_page_type_choice', array('required' => false))
+                    ->add(
+                        'type',
+                        method_exists('Symfony\Component\Form\AbstractType', 'getBlockPrefix') ?
+                            'Sonata\PageBundle\Form\Type\PageTypeChoiceType' :
+                            'sonata_page_type_choice',
+//                        'Sonata\PageBundle\Form\Type\PageTypeChoiceType',
+                        array('required' => false))
                 ->end()
             ;
         }
 
+        /*
+         * NEXT_MAJOR: remove method_exists and uncomment the second type when dropping sf < 2.8
+         */
         $formMapper
             ->with('form_page.group_main_label')
-                ->add('templateCode', 'sonata_page_template', array('required' => true))
+                ->add(
+                    'templateCode',
+                    method_exists('Symfony\Component\Form\AbstractType', 'getBlockPrefix') ?
+                        'Sonata\PageBundle\Form\Type\TemplateChoiceType' :
+                        'sonata_page_template',
+//                    'Sonata\PageBundle\Form\Type\TemplateChoiceType',
```

Comment:
There's a NEXT_MAJOR comment a couple of lines above that explains this. That's the pattern used on Issue #820. If this is wrong, give me a good example to follow.